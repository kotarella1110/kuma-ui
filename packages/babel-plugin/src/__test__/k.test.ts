import { babelTransform } from "./testUtils";

describe("k api", () => {
  describe("Snapshot tests", () => {
    test("basic usage should match snapshot", () => {
      // Arrange
      const inputCode = `
        import { k } from '@kuma-ui/core'
        function App() {
          return <k.div fontSize={24}></k.div>
        }
      `;
      // Act
      const { code } = babelTransform(inputCode);
      // Assert
      expect(code).toMatchSnapshot();
    });

    test("using responsive props should match snapshot", () => {
      // Arrange
      const inputCode = `
        import { k } from '@kuma-ui/core'
        function App() {
          return <k.a fontSize={[16, 24]} />
        }
      `;
      // Act
      const { code } = babelTransform(inputCode);
      // Assert
      expect(code).toMatchSnapshot();
    });

    test("using space props should match snapshot", () => {
      // Arrange
      const inputCode = `
        import { k } from '@kuma-ui/core'
        function App() {
          return <k.div p={2} />
        }
      `;
      // Act
      const { code } = babelTransform(inputCode);
      // Assert
      expect(code).toMatchSnapshot();
    });

    test("using pseudo elements should match snapshot", () => {
      // Arrange
      const inputCode = `
        import { k } from '@kuma-ui/core'
        function App() {
          return <k.div p={2} _after={{ color: 'blue' }} />
        }
      `;
      // Act
      const { code } = babelTransform(inputCode);
      // Assert
      expect(code).toMatchSnapshot();
    });

    test("using pseudo props should match snapshot", () => {
      // Arrange
      const inputCode = `
        import { k } from '@kuma-ui/core'
        function App() {
          return <k.span p={2} _hover={{ color: 'red' }} />
        }
      `;
      // Act
      const { code } = babelTransform(inputCode);
      // Assert
      expect(code).toMatchSnapshot();
    });
  });
});
