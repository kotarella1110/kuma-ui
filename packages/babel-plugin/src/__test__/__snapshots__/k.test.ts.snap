// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`k api > Snapshot tests (runtime: automatic) > basic usage should match snapshot 1`] = `
"
.kuma-1119741946 {font-size:24px;}

import React from \\"react\\";
import { jsx as _jsx } from \\"react/jsx-runtime\\";
function App() {
  return /*#__PURE__*/_jsx(\\"div\\", {
    className: [\\"kuma-1119741946\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: automatic) > using pseudo elements should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}.kuma-110275621:after{color:blue;}

import React from \\"react\\";
import { jsx as _jsx } from \\"react/jsx-runtime\\";
function App() {
  return /*#__PURE__*/_jsx(\\"div\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: automatic) > using pseudo props should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}.kuma-110275621:hover{color:red;}

import React from \\"react\\";
import { jsx as _jsx } from \\"react/jsx-runtime\\";
function App() {
  return /*#__PURE__*/_jsx(\\"span\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: automatic) > using responsive props should match snapshot 1`] = `
"
.kuma-4054065355 {font-size:16px;}@media (min-width:576px){.kuma-4054065355{font-size:24px;}}

import React from \\"react\\";
import { jsx as _jsx } from \\"react/jsx-runtime\\";
function App() {
  return /*#__PURE__*/_jsx(\\"a\\", {
    className: [\\"kuma-4054065355\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: automatic) > using space props should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}

import React from \\"react\\";
import { jsx as _jsx } from \\"react/jsx-runtime\\";
function App() {
  return /*#__PURE__*/_jsx(\\"div\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: classic) > basic usage should match snapshot 1`] = `
"
.kuma-1119741946 {font-size:24px;}

import React from \\"react\\";
function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: [\\"kuma-1119741946\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: classic) > using pseudo elements should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}.kuma-110275621:after{color:blue;}

import React from \\"react\\";
function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: classic) > using pseudo props should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}.kuma-110275621:hover{color:red;}

import React from \\"react\\";
function App() {
  return /*#__PURE__*/React.createElement(\\"span\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: classic) > using responsive props should match snapshot 1`] = `
"
.kuma-4054065355 {font-size:16px;}@media (min-width:576px){.kuma-4054065355{font-size:24px;}}

import React from \\"react\\";
function App() {
  return /*#__PURE__*/React.createElement(\\"a\\", {
    className: [\\"kuma-4054065355\\"].join(\\" \\")
  });
}
"
`;

exports[`k api > Snapshot tests (runtime: classic) > using space props should match snapshot 1`] = `
"
.kuma-110275621 {padding:2px;}

import React from \\"react\\";
function App() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    className: [\\"kuma-110275621\\"].join(\\" \\")
  });
}
"
`;
